File: Maratona_Java/src/OBI2014/triangulo.java
/* Guilherme A. Pinto - OBI 2014 - Triangulo */

import java.util.Scanner;
import java.io.*;

public class guilherme_java {

    public static boolean triangulo( int p, int q, int r  ){
	if ( p < q+r && q < p+r && r < p+q ) return true;
	return false;
    }

    public static void main(String[] args) {
	Scanner in = new Scanner(System.in);
	int a,b,c,d;

	a = in.nextInt();
	b = in.nextInt();
	c = in.nextInt();
	d = in.nextInt();

	if ( triangulo( a, b, c ) ||
	     triangulo( a, b, d ) ||
	     triangulo( a, c, d ) ||
	     triangulo( b, c, d ) ) System.out.print("S\n");
	else  System.out.print("N\n");
    }
}

File: Maratona_Python/OBI2014/triangulo.py
#!/usr/bin/env python2.7
# encoding: utf-8

# Guilherme A. Pinto - OBI 2014 - Triangulo

import sys

def get_linha_num(dados):
    for i in range(len(dados)):
        dados[i] = int(dados[i])
    return dados

def triangulo( p, q, r ):
    if ( p < q+r and q < p+r and r < p+q ): return True
    return False

finput = sys.stdin
linha = finput.readline()[:-1]

a,b,c,d = get_linha_num(linha.split(" "))

if ( triangulo( a, b, c ) or
     triangulo( a, b, d ) or
     triangulo( a, c, d ) or
     triangulo( b, c, d ) ):
    print ("S")
else:
    print ("N")

sys.exit(0)
